server/controllers/commentController.js

```diff
@@ -0,0 +1,208 @@
+      return res.status(201).send({
+        success: true,
+        message: 'Comment created successfully',
+        comment: commentCreated
+      });
+    } catch (error) {
+      return next(error);
+    }
+  }
+
+  /**
+   * @description Get all Comments for a particular article
+   * @param {object} req - req from route
+   * @param {object} res - respose to route
+   * @param {object} next - callback function
+   * @returns {object} a response object
+   */
+  static async getArticleComments(req, res, next) {
+    const { params:{articleId}} = req;
+    try {
+      const comments = await Comment.findAndCountAll(
+        {
+          include: [{
+            model: User,
+            as: 'author',
+            attributes: ['userName', 'imageUrl', 'bio']
+
+          }],
+          attributes: { exclude: ['userId'] },
+          where: { articleId }
+        });
+
+      if (comments.count == 0) {
+        return res.status(404).json({
+          success: false,
+          message: 'No Comment for this Article',
+        });
+      }
+
+      return res.status(200).json({
+        success: true,
+        message: 'Retrieved Comments successfully',
+        comments
+      });
+
+    } catch (error) {
+      return next(error);
+    }
+  }
+
+  /**
+   * @param {object} req - req from route
+   * @param {object} res - respose to route
+   * @param {object} next - callback function
+   * @returns {object} a response object
+   */
+  static async getSingleComment(req, res, next) {
+    const { params: { id }} = req;
+    try {
+      const comment = await Comment.findOne(
+        {
+          include: [{
+            model: User,
+            as: 'author',
+            attributes: ['userName', 'imageUrl', 'bio']
+
+          }],
+          attributes: { exclude: ['userId'] },
+          where: { id }
+        });
+
+      if (!comment) {
+        return res.status(404).json({
+          success: false,
+          message: 'No Comment found',
+        });
+      }
+
+      return res.status(200).json({
+        success: true,
+        message: 'Retrieved comment successfully',
+        comment
+      });
+
+    } catch (error) {
+      return next(error);
+    }
+  }
+
+  /**
+   * @description only owners of comment should be able to update
+   * @param {object} req - req from route
+   * @param {object} res - respose to route
+   * @param {object} next - callback function
+   * @returns {object} a response object
+   */
+  static async updateComment(req, res, next) {
+    const { user:{ id },body:{ comment }, params:{ commentId }} = req;
+    try {
+      const checkCommentOwner = await Comment.findOne({
+        where: {
+          id: commentId,
+          userId: id,
+         }
+      });
+      if (!checkCommentOwner) {
+        return res.status(403).json({
+          success: false,
+          message: 'User cannot update comment they did\'t create',
+        });
+      }
+      await Comment.update(
```

Comment:
Lets write this in a more concise form.

Reply:
updated to `User is not authorized to update comment`